blackboxopt/optimizers/botorch_base.py

```diff
@@ -44,6 +45,21 @@
     ) from e
 
 
+def _get_numerical_points_from_discrete_space(space: ParameterSpace) -> np.ndarray:
+    """Retrieve all points from a discrete space in the numerical representation"""
+    points_along_dimensions = []
+    for parameter_name in space.get_parameter_names():
+        num_values = space.get_parameter_by_name(parameter_name)[
+            "parameter"
+        ].num_values  # type:ignore
+        points_along_dimensions.append(
+            np.linspace(1 / (2 * num_values), 1 - 1 / (2 * num_values), num_values)
+        )
+    points = np.meshgrid(*points_along_dimensions)
+    points = [p.reshape((p.size, 1)) for p in points]
+    return np.concatenate(points, axis=-1)
+
+
 def _acquisition_function_optimizer_factory(
```

Comment:
Let's add another sentence to the function docstring explaining that e.g. "In case an exclusively discrete space is detected and `num_random_choices` is not specified, the discrete acquisition function optimizer is using all possible combinations in the discrete space."